src/ReactiveUI/Platforms/windows-common/ViewModelViewHost.cs

```diff
@@ -106,15 +105,17 @@ public ViewModelViewHost()
                 .StartWith(platformGetter())
                 .DistinctUntilChanged();
 
-            var contractChanged = _updateViewModel.Select(_ => ViewContractObservable).Switch();
-            var viewModelChanged = _updateViewModel.Select(_ => ViewModel);
+            // Observable that fires when ViewModel or ViewContractObservable change, or ViewContractObservable fires
+            var vmAndContract = Observable.CombineLatest(
```

Comment:
This is going to cause memory leaks unfortunately due to the way dependency properties work. Eg WhenAnyValue will call into the static manager, and it'll hold an instance of the class until you dispose the WhenAnyValue.

This is the reason why the SomethingChanged exists since it doesn't suffer from this problem.

Happy to take a solution that keeps the SomethingChanged approach.

Reply:
Did not know that could be an issue. Is that not a problem in https://github.com/reactiveui/ReactiveUI/blob/main/src/ReactiveUI.XamForms/ViewModelViewHost.cs as well then?

Reply:
Xamarin Forms is different since they have BindableProperty invoke a INotifyPropertyChanged so have a non-memory leaking way of getting alerted of property changes.